pallets/settlement/src/benchmarking.rs

```diff
@@ -216,33 +231,345 @@ fn emulate_add_instruction<T: Trait>(
+}
+
+fn setup_affirm_instruction<T: Trait>(
+    l: u32,
+) -> (
+    Vec<PortfolioId>,
+    User<T>,
+    User<T>,
+    Ticker,
+    Ticker,
+    Vec<Leg<T::Balance>>,
+) {
+    // create venue
+    let from = UserBuilder::<T>::default().generate_did().build("creator");
+    let venue_id = create_venue_::<T>(from.did(), vec![]);
+    let settlement_type: SettlementType<T::BlockNumber> = SettlementType::SettleOnAffirmation;
+    let to = UserBuilder::<T>::default().generate_did().build("receiver");
+    let mut portfolios_from: Vec<PortfolioId> = Vec::with_capacity(l as usize);
+    let mut portfolios_to: Vec<PortfolioId> = Vec::with_capacity(l as usize);
+    let mut legs: Vec<Leg<T::Balance>> = Vec::with_capacity(l as usize);
+
+    // Create legs vector.
+    // Assuming the worst case where there is no dedup of `from` and `to` in the legs vector.
+    // Assumption here is that instruction will never be executed as still there is one auth pending.
+    let from_ticker = make_base_asset::<T>(
+        &from,
+        true,
+        Some(Ticker::try_from(vec![b'A'; 8 as usize].as_slice()).unwrap()),
+    );
+    let to_ticker = make_base_asset::<T>(
+        &to,
+        true,
+        Some(Ticker::try_from(vec![b'B'; 8 as usize].as_slice()).unwrap()),
+    );
+    for n in 0..l / 2 {
+        setup_leg_and_portfolio_with_ticker::<T>(
+            Some(to.did()),
+            Some(from.did()),
+            from_ticker,
+            to_ticker,
+            n,
+            &mut legs,
+            &mut portfolios_from,
+            &mut portfolios_to,
+        );
+    }
+    Module::<T>::add_and_affirm_instruction(
+        (from.origin.clone()).into(),
+        venue_id,
+        settlement_type,
+        None,
+        legs.clone(),
+        portfolios_from,
+    )
+    .expect("Unable to add and affirm the instruction");
+    (portfolios_to, from, to, from_ticker, to_ticker, legs)
+}
+
+fn add_smart_extension_to_ticker<T: Trait>(
+    code_hash: <T::Hashing as Hash>::Output,
+    origin: RawOrigin<T::AccountId>,
+    account: T::AccountId,
+    ticker: Ticker,
+) {
+    let data = vec![
+        209, 131, 81, 43, 160, 134, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
+    ]; // Allow 100% as percentage ownership and allow primary issuance.
+    <polymesh_contracts::Module<T>>::instantiate(
+        origin.clone().into(),
+        0.into(),
+        Weight::max_value(),
+        code_hash,
+        data.clone(),
+        0.into(),
+    )
+    .expect("Settlement: Failed to instantiate the contract");
+    let extension_id =
+        T::DetermineContractAddress::contract_address_for(&code_hash, &data, &account);
+    let extension_details = SmartExtension {
+        extension_type: SmartExtensionType::TransferManager,
+        extension_name: b"PTM".into(),
+        extension_id: extension_id.clone(),
+        is_archive: false,
+    };
+    <pallet_asset::Module<T>>::add_extension(origin.into(), ticker, extension_details)
```

Comment:
For the worst case, the smart extension should consume 100% of the limit. This does not.

Reply:
I don't get it??

Reply:
Let's just close it since we are ditching SE.